main.go

```diff
@@ -261,6 +265,74 @@ func writeNewXWithOptionsAndDefaults(buf *jen.File, c Config) {
 	})
 }
 
+var (
+	DebugMapFieldTag = "debugmap"
+	SensitiveNames   = []string{"secure"}
+)
+
+func writeDebugMap(buf *jen.File, st *types.Struct, c Config) {
+	newFuncName := fmt.Sprintf("DebugMap")
+
+	buf.Comment(fmt.Sprintf("%s returns a map form of %s for debugging", newFuncName, c.TargetTypeName))
+	buf.Func().Params(jen.Id(c.ReceiverId).Id(c.StructName)).Id(newFuncName).Params().Id("map[string]any").BlockFunc(func(grp *jen.Group) {
+		mapId := "debugMap"
+		grp.Id(mapId).Op(":=").Map(jen.String()).Any().Values()
+
+		for i := 0; i < st.NumFields(); i++ {
+			f := st.Field(i)
+			if f.Anonymous() || !f.Exported() {
+				continue
+			}
+
+			tags, err := structtag.Parse(st.Tag(i))
+			if err != nil {
+				panic(err)
+			}
+
+			tag, err := tags.Get(DebugMapFieldTag)
+			if err != nil {
+				fmt.Printf("missing debugmap tag on field %s in type %s\n", f.Name(), c.TargetTypeName)
+				os.Exit(1)
+			}
+
+			switch tag.Value() {
+			case "visible":
+				for _, sensitiveName := range SensitiveNames {
+					if strings.Contains(strings.ToLower(f.Name()), sensitiveName) {
+						fmt.Printf("field %s in type %s must be marked as 'sensitive'\n", f.Name(), c.TargetTypeName)
+						os.Exit(1)
+					}
+				}
+
+				grp.Id(mapId).Index(jen.Lit(f.Name())).Op("=").Qual("github.com/ecordell/optgen/helpers", "DebugValue").Call(jen.Id(c.ReceiverId).Dot(f.Name()), jen.Lit(false))
+
+			case "visible-format":
+				for _, sensitiveName := range SensitiveNames {
+					if strings.Contains(strings.ToLower(f.Name()), sensitiveName) {
```

Comment:
this should probably be configurable via the cli args - not everyone will necessarily agree on sensitivenames